trino-aws-proxy/src/main/java/io/trino/aws/proxy/server/rest/S3PresignController.java

```diff
@@ -53,11 +62,25 @@ public Map<String, URI> buildPresignedRemoteUrls(SigningMetadata signingMetadata
     {
         Optional<Instant> signatureExpiry = Optional.of(Instant.now().plusMillis(presignUrlDuration.toMillis()));
 
-        return Stream.of("GET", "PUT", "POST", "DELETE")
+        return getPresignedUrlRequestedMethods(request).stream()
                 .flatMap(httpMethod -> buildPresignedRemoteUrl(httpMethod, signingMetadata, request, targetRequestTimestamp, remoteUri, signatureExpiry))
                 .collect(toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
     }
 
+    private Set<String> getPresignedUrlRequestedMethods(ParsedS3Request request)
+    {
+        Set<String> requestedMethods = request.requestHeaders().unmodifiedHeaders().get(PRESIGNED_URL_METHODS_HEADER_NAME).stream()
```

Comment:
`PresignAwareAmazonS3` and `PresignAwareS3Client` need to be updated with this new header.